components/raftstore/src/store/fsm/apply.rs

```diff
@@ -1029,15 +1033,18 @@ where
                 break;
             }
 
-            let expect_index = self.apply_state.get_applied_index() + 1;
-            if expect_index != entry.get_index() {
-                panic!(
-                    "{} expect index {}, but got {}, ctx {}",
-                    self.tag,
-                    expect_index,
-                    entry.get_index(),
-                    apply_ctx.tag,
-                );
+            let peer = util::find_peer_by_id(&self.region, self.id).unwrap();
+            if !peer.is_witness && !self.wait_data {
```

Comment:
we should stop applying following commands when `wait_data` is set, until the snapshot is applied

Reply:
Should also consider `ready_to_handle_pending_snap`

Reply:
> we should stop applying following commands when `wait_data` is set, until the snapshot is applied

It is currently detected that `wait_data` is `true`, return `yeild` until the apply snapshot is completed.

> Should also consider ready_to_handle_pending_snap

Sure, if `wait_data` is detected, the index judgment condition is ignored, as apply fsm has been in yield state.